Extensions/shuffle+.js

```diff
@@ -1,148 +1,269 @@
-// @ts-check
-
 // NAME: Shuffle+
-// AUTHOR: khanhas
+// AUTHOR: khanhas, Tetrax-10
 // DESCRIPTION: True shuffle with no bias.
 
 /// <reference path="../globals.d.ts" />
 
-(function ShufflePlus() {
-    if (!Spicetify.CosmosAsync || !Spicetify.Platform) {
-        setTimeout(ShufflePlus, 1000);
+let shufflePlusCount = 0;
+(async function shufflePlus() {
+    if (!(Spicetify.CosmosAsync && Spicetify.Platform && shufflePlusCount < 200)) {
+        setTimeout(shufflePlus, 300);
+        shufflePlusCount++;
         return;
```

Comment:
Why do you need the function to recursively run 200 times before it is executed?

Reply:
this is how ext works

Reply:
> it does't run for 200 times. 200 is the limit, prob spicetify will be available with in 3 to 10 loops. if spicetify is not available then this function will loop infinite no of time. so i set a limit 200

If both `Spicetify.Platform` and `Spicetify.CosmosAsync` is not available, this is a major CLI bug that needs to be patched. This is not required.

Reply:
> it does't run for 200 times. 200 is the limit.

The function would continue to execute as normal when the counter reaches 200 even when both `Platform` and `CosmosAsync` are not available, which breaks the extension anyway and defeats the purpose of the null check
![image](https://user-images.githubusercontent.com/77577746/189112093-0a8719f9-a16f-429c-a0ed-78ef2686ab1b.png)

Reply:
> this is my way of writing if you don't like it. you change whatever you want am not gonna reject any changes from now. i have my version with me, so its up to you.

It's not a matter of style, this is actually crucial to the extension. I'm not the antagonist here nor am I trying to nitpick or reject your work, these are constructive suggestions with explanations attached that helps you understand my decision on why I commented on it. Please do understand that.

Reply:
I do apologize if I gave such impression and hope that you understand my intentions here.

Reply:
> 

fixed it

Reply:
6 seconds ?? 300x200 = 60 seconds

```js
let shufflePlusCount = 0;
(async function shufflePlus() {
    if (!(Spicetify.CosmosAsync && Spicetify.Platform) && shufflePlusCount < 200) {
        setTimeout(shufflePlus, 300);
        shufflePlusCount++;
       console.log(shufflePlusCount); // add this line and run
        return;
    }
    await initShufflePlus();
})();
```

Reply:
This counter is just an extra layer to prevent bugs nothing wrong in that.

For example in `Skip or Play Liked Songs` extension if no audio device is connected the extension will loop infinite times because `Platform ` and `LocalStorage` will be available but not `Player`. so 60 sec is the limit if no audio device is connected then it will stop init process
![image](https://user-images.githubusercontent.com/75513645/189123320-c06843a2-ce84-498f-8111-ec01140a7e4c.png)

here for eg lets assume `CosmosAsync` hook broke i dont want this init process to run infinite times. so gave 60 sec time limit

Reply:
> This counter is just an extra layer to prevent bugs nothing wrong in that.

Shuffle+ would not execute with either of those hooks broke. The bugs wouldn't even be found unless the CLI is patched as mentioned before.

> here for eg lets assume `CosmosAsync` hook broke i dont want this init process to run infinite times. so gave 60 sec time limit.

If either of those hooks broke, none of Spicetify functionality (except for themes) would work, and stopping the recursive initialization would not bring any significant benefit except for adding another unnecessary and confusing condition to those who review the extension.

> 6 seconds ?? 300x200 = 60 seconds

My bad, point still stands.